clang/lib/CodeGen/CGOpenMPRuntime.cpp

```diff
@@ -2331,8 +2332,11 @@ void CGOpenMPRuntime::emitBarrierCall(CodeGenFunction &CGF, SourceLocation Loc,
   auto *OMPRegionInfo =
       dyn_cast_or_null<CGOpenMPRegionInfo>(CGF.CapturedStmtInfo);
   if (CGF.CGM.getLangOpts().OpenMPIRBuilder) {
-    CGF.Builder.restoreIP(OMPBuilder.createBarrier(
-        CGF.Builder, Kind, ForceSimpleCall, EmitChecks));
+    llvm::OpenMPIRBuilder::InsertPointOrErrorTy AfterIP =
+        OMPBuilder.createBarrier(CGF.Builder, Kind, ForceSimpleCall,
+                                 EmitChecks);
+    assert(AfterIP && "unexpected error creating barrier");
```

Comment:
[serious] @skatrak `assert` is active by `LLVM_ENABLED_ASSERTS`, but `llvm::Error`/`llvm::Expected` uses `LLVM_ABI_BREAKING_CHECKS`. In the configuration `-DLLVM_ENABLED_ASSERTS=OFF -DLLVM_ABI_BREAKING_CHECKS=FORCE_ON` the error check is not done, but then crashes unconditional to about unchecked errors.

Reply:
I guess then these assert should be replaced with `if (!AfterIP) llvm_unreachable(...)`, or is there a better way to deal with this?

Reply:
Here's the PR to address this issue: #115863. There was indeed a cleaner solution to this.